server/middleware/logging.go

```diff
@@ -25,14 +25,17 @@ func withLogging(
 
 		args := []any{
 			"method", r.Method, "url", r.URL,
-			"status", scw.status, "duration", time.Since(start),
 			"commitment_mode", mode, "cert_version", getCertVersion(r),
+			"status", scw.status, "duration", time.Since(start),
 		}
 
 		if err != nil {
 			args = append(args, "error", err.Error())
 			log.Error("request completed with error", args...)
 		} else {
+			// This log line largely duplicates the logging in the handlers.
+			// Only difference being that we have duration here, whereas the handlers log the cert.
+			// TODO: should we also pass the cert via the requestContext to rid of the log lines in the handlers?
```

Comment:
Is the idea here that we would only use the cert for logging here? Or would this be a broader change to how we pass the cert around?

Reply:
Yes purely for middleware use. I don't know if its needed if we use request IDs though... have to think about this whole thing more. Feels fishy to have to pass a cert via context just for logging I admit.